instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -245,11 +256,91 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
         assertEquals(277.0, alternativesMetadata.infoFromStartOfPrimary.duration, 1.0)
     }
 
+    @Test
+    fun expect_route_refresh_to_update_annotations_for_truncated_current_leg() =
+        sdkTest {
+            setupMockRequestHandlers(
+                twoCoordinates,
+                R.raw.route_response_route_refresh,
+                R.raw.route_response_route_refresh_truncated_first_leg,
+                "route_response_route_refresh",
+                acceptedGeometryIndex = 5
+            )
+            val routeOptions = generateRouteOptions(twoCoordinates)
+            val requestedRoutes = mapboxNavigation.requestRoutes(routeOptions)
+                .getSuccessfulResultOrThrowException()
+                .routes
+
+            mapboxNavigation.setNavigationRoutes(requestedRoutes)
+            mapboxNavigation.startTripSession()
+            // corresponds to currentRouteGeometryIndex = 5
+            stayOnPosition(38.57622, -121.496731)
```

Comment:
@mapbox/navnative , do you expect us jumping from initial position(-121.496066, 38.577764) to the one needed for test in one step? Is it okay if we carry on doing this? Should be expect any consequences of this jump?